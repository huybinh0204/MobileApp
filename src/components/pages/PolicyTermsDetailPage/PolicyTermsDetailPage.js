import React from 'react';
import { WebView } from 'react-native-webview';
import { SecondaryHeader } from '_components/organisms';
import { COLORS } from '_constants';
import strings from '_localization/index';
import { ActivityIndicator, Container } from './PolicyTermsDetailPage.styles';

const PolicyTermsDetailPage = ({ url }) => {
  return (
    <Container testID="PolicyTermsDetailPage">
      <SecondaryHeader title={strings.settings.policyTerms.legalDocument} />
      <WebView
        overScrollMode="always"
        renderLoading={() => <ActivityIndicator color={COLORS.alpha500} size="large" />}
        source={{ uri: url }}
        startInLoadingState={true}
      />
    </Container>
  );
};

export default PolicyTermsDetailPage;
