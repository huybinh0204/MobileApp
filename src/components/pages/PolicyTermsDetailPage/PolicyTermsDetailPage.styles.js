import styled from '@emotion/native';

export const Container = styled.SafeAreaView(({ theme }) => ({
  flex: 1,
  alignContent: 'center',
  backgroundColor: theme.colors.white,
}));

export const ActivityIndicator = styled.ActivityIndicator(({ theme }) => ({
  flex: 1,
  justifyContent: 'flex-start',
  backgroundColor: theme.colors.white,
}));
