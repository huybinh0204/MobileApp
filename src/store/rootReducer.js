import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
