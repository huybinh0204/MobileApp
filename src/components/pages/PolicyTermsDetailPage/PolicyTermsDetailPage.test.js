import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import { createStore } from 'redux';
import PolicyTermsDetailPage from './PolicyTermsDetailPage';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';

describe('PolicyTermsDetailPage', () => {
  it('should match snapshot', async () => {
    const mockStore = createStore(rootReducer, {});

    const { toJSON } = render(
      <ComponentWithProviders component={PolicyTermsDetailPage} store={mockStore} />
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
