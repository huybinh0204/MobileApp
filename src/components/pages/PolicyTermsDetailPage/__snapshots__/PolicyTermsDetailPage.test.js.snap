// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyTermsDetailPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "alignContent": "center",
                              "backgroundColor": "#FFFFFF",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="PolicyTermsDetailPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            Legal Document
                          </Text>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "overflow": "hidden",
                              },
                              undefined,
                            ]
                          }
                        >
                          <RNCWebView
                            cacheEnabled={true}
                            injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                            injectedJavaScriptForMainFrameOnly={true}
                            javaScriptEnabled={true}
                            messagingEnabled={false}
                            onContentProcessDidTerminate={[Function]}
                            onHttpError={[Function]}
                            onLoadingError={[Function]}
                            onLoadingFinish={[Function]}
                            onLoadingProgress={[Function]}
                            onLoadingStart={[Function]}
                            onMessage={[Function]}
                            onShouldStartLoadWithRequest={[Function]}
                            overScrollMode="always"
                            source={
                              Object {
                                "uri": undefined,
                              }
                            }
                            startInLoadingState={true}
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "overflow": "hidden",
                                },
                                Object {
                                  "backgroundColor": "#ffffff",
                                },
                                undefined,
                              ]
                            }
                            textInteractionEnabled={true}
                            useSharedProcessPool={true}
                          />
                          <ActivityIndicator
                            color="#3D3397"
                            size="large"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#FFFFFF",
                                  "flex": 1,
                                  "justifyContent": "flex-start",
                                },
                                undefined,
                              ]
                            }
                          />
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
